fix: Display Media Device labels for Firefox

EnricoSchw opened the pull request:
## Description
In Firefox, the labels of the Media devices can only be read under certain constrains. This means that the device names cannot be displayed in the select menu. This PR is a fix for it

## Screenshots/Screencast (for UI changes)
Before this fix, the selection fields are displayed with blank labels or common names:

<img width="728" alt="Bildschirmfoto 2024-06-12 um 14 06 46" src="https://github.com/wireapp/wire-webapp/assets/1362436/90eb7775-88e8-4530-b561-bce08a52a0ed">

<img width="1031" alt="Bildschirmfoto 2024-06-12 um 18 21 16" src="https://github.com/wireapp/wire-webapp/assets/1362436/c9762ba2-2990-429f-b213-8fc4e30a6b4b">

----

With this fix, the selection fields are displayed with device names:

<img width="860" alt="Bildschirmfoto 2024-06-12 um 16 34 24" src="https://github.com/wireapp/wire-webapp/assets/1362436/183e794a-6c90-4667-8254-51d08ed54a6d">

<img width="1029" alt="Bildschirmfoto 2024-06-12 um 18 21 58" src="https://github.com/wireapp/wire-webapp/assets/1362436/52283b53-4934-449d-aedc-2e26460dbbe9">

----

# Notes
The fix has one disadvantage. I had to change the device list initialization process to avoid asking a user for audio access rights on first login starting page.

![Bildschirmfoto 2024-06-24 um 15 06 23](https://github.com/wireapp/wire-webapp/assets/1362436/5c51ec4b-8e98-4dfc-80ae-b0ca4ffd5d04)

In order to display the correct device labels, we need to make an access authorization query when creating the device list. But we should not ask for microphone and camera access when not needed. 

To avoid a permission query being displayed when starting the app, I have rewritten the `MediaDeviceHandler` and The device list is no longer created in the constructor. The method [`MediaDeviceHandler.initializeMediaDevices`](https://github.com/wireapp/wire-webapp/blob/58675eaa1ab85f2553ede4d49d875991b2627c94/src/script/media/MediaDevicesHandler.ts#L168-L176) is now integrated into the three points in our app where a call can be accepted or started. In addition, this method is also called in the device settings. So, a device list is only created before the first call or when the settings are called. 

## Checklist

- [ ] PR has been self reviewed by the author;
- [ ] Hard-to-understand areas of the code have been commented;
- [ ] If it is a core feature, unit tests have been added;

### Important details for the reviewers

(Delete this section if unnecessary)

- use (x) data
- can be reviewed commit-by-commit
- be sure to look at ...

EnricoSchw committed:
authorized to access the devices labels by calling getUserMedia

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/wireapp/wire-webapp?pullRequest=17562) <br/>All committers have signed the CLA.

tlebon commented:
does this change anything for chromium browsers?

tlebon commented:
thanks for taking the initiative on this!

EnricoSchw commented:
@tlebon Thx for your fast response

Unfortunately it also has an impact on Chrome. The device lists are created during the initialization process of the app, so that the access popup also appears in Chrome. 

I see three solutions here:
1. A browser switch for Firefox (could be a temporary solution)
2. Only have a numbered device list for Firefox
3. Only initialize the device list when it is needed

What do you think?

More detail:
Some users will click away the pop-up and not allow access. When the mic is needed for a call, the pop-up will no longer appear and the mic will be blocked. We will receive many support requests because the mic is not working.
This is very bad UX and that's why I marked the PR as draft.

WolfiWire commented:
Do we have the possibility to create a custom dialog (that explains better why we are asking for these access rights), or are we bound to the browser default popup?

EnricoSchw commented:
@WolfiWire I would use this one.

![Bildschirmfoto 2024-06-14 um 10 17 12](https://github.com/wireapp/wire-webapp/assets/1362436/71036d9a-8b61-473b-967d-030c24323794)

EnricoSchw committed:
fet: init devices in settings and only when needed to avoid to much permission checks

EnricoSchw committed:
fet: requesting media device list before entering a call

  - Now the devices are only queried before a call or when the settings are opened.

EnricoSchw committed:
fet: repair and adjust existing tests

EnricoSchw commented:
@WolfiWire I have rewritten the device list initialization. This means that our app behaves the same for the user as before. Only internally the process is different.

EnricoSchw commented:
@tlebon Please, could someone take a look at this?

tlebon commented:
I'm still OoO (sick). Maybe @wireapp/web can take a look. Or else I will tomorrow.

EnricoSchw commented:
> I'm still OoO (sick). 

Get well soon!

> Maybe @wireapp/web can take a look. Or else I will tomorrow.

Thx. sometime this week would be enough
